streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -212,6 +212,11 @@
      */
     public static final String OPTIMIZE = "all";
 
+    /**
+     * Config value for parameter (@link #TOPOLOGY_OPTIMIZATION "topology.optimization" for enabling topology optimization
+     */
+    public static final String OPTIMIZE_AT_20 = "2.0";
```

Comment:
Why do we need this? Should we not use `"all"` ? Also, this would be  public API change -- we can still update the corresponding KIP, but I don't think we need this, do we?

Reply:
I agree.  Since this our first optimization, `"all"` should be sufficient.  But this raises another good point if we plan to track optimization versions are we setting ourselves up for a KIP for each optimization release?

Reply:
1. Sounds good. I'm reverting to use `all`.
2. Moving forward, not needing one KIP for each release. We can use a single KIP adding the mechanism, and if two versions are compatible (say if no new incompatible optimizations introduced from X to Y), we can just document it in upgrade section.